File: internal/model/connection.go
// Package model 定义应用程序的数据模型
package model

// ConnectionEnv 连接环境类型
type ConnectionEnv string

const (
	EnvProduction  ConnectionEnv = "生产"
	EnvTest        ConnectionEnv = "测试"
	EnvDevelopment ConnectionEnv = "开发"
)

// ConnectionStatus 连接状态
type ConnectionStatus string

const (
	StatusOnline  ConnectionStatus = "online"
	StatusOffline ConnectionStatus = "offline"
)

// Connection 连接配置
type Connection struct {
	ID         int              `json:"id"`         // 连接ID
	Name       string           `json:"name"`       // 连接名称
	Env        ConnectionEnv    `json:"env"`        // 环境类型
	NameServer string           `json:"nameServer"` // NameServer 地址
	TimeoutSec int              `json:"timeoutSec"` // 超时时间(秒)
	EnableACL  bool             `json:"enableACL"`  // 是否启用 ACL 认证
	AccessKey  string           `json:"accessKey"`  // ACL AccessKey
	SecretKey  string           `json:"secretKey"`  // ACL SecretKey
	Status     ConnectionStatus `json:"status"`     // 连接状态
	LastCheck  string           `json:"lastCheck"`  // 最近检测时间
	IsDefault  bool             `json:"isDefault"`  // 是否默认连接
	Remark     string           `json:"remark"`     // 备注
}

File: frontend/src/api/connection.ts
import * as ConnectionService from '../../bindings/rocket-leaf/internal/service/connectionservice.js'
import type { Connection } from '../../bindings/rocket-leaf/internal/model/models.js'

export async function getConnections(): Promise<(Connection | null)[]> {
  try {
    return await ConnectionService.GetConnections()
  } catch (e) {
    console.error('GetConnections', e)
    throw e
  }
}

export async function addConnection(
  name: string,
  env: string,
  nameServer: string,
  timeoutSec: number,
  enableACL: boolean,
  accessKey: string,
  secretKey: string,
  remark: string,
): Promise<Connection | null> {
  try {
    return await ConnectionService.AddConnection(
      name,
      env,
      nameServer,
      timeoutSec,
      enableACL,
      accessKey,
      secretKey,
      remark,
    )
  } catch (e) {
    console.error('AddConnection', e)
    throw e
  }
}

export async function updateConnection(
  id: number,
  name: string,
  env: string,
  nameServer: string,
  timeoutSec: number,
  enableACL: boolean,
  accessKey: string,
  secretKey: string,
  remark: string,
): Promise<Connection | null> {
  try {
    return await ConnectionService.UpdateConnection(
      id,
      name,
      env,
      nameServer,
      timeoutSec,
      enableACL,
      accessKey,
      secretKey,
      remark,
    )
  } catch (e) {
    console.error('UpdateConnection', e)
    throw e
  }
}

export async function deleteConnection(id: number): Promise<void> {
  try {
    await ConnectionService.DeleteConnection(id)
  } catch (e) {
    console.error('DeleteConnection', e)
    throw e
  }
}

export async function connect(id: number): Promise<void> {
  try {
    await ConnectionService.Connect(id)
  } catch (e) {
    console.error('Connect', e)
    throw e
  }
}

export async function disconnect(id: number): Promise<void> {
  try {
    await ConnectionService.Disconnect(id)
  } catch (e) {
    console.error('Disconnect', e)
    throw e
  }
}

export async function connectDefault(): Promise<void> {
  try {
    await ConnectionService.ConnectDefault()
  } catch (e) {
    console.error('ConnectDefault', e)
    throw e
  }
}

export async function setDefaultConnection(id: number): Promise<void> {
  try {
    await ConnectionService.SetDefaultConnection(id)
  } catch (e) {
    console.error('SetDefaultConnection', e)
    throw e
  }
}

export async function testConnection(id: number): Promise<string> {
  try {
    return await ConnectionService.TestConnection(id)
  } catch (e) {
    console.error('TestConnection', e)
    throw e
  }
}
